### PR TITLE
US136048: Display One Drive icon that prompts to open in new window

### PR DESCRIPTION
**Pull request by coffeexcode**
## Relevant Rally
- [US136048](https://rally1.rallydev.com/#/?detail=/userstory/626774509603&fdp=true): Display One Drive icon that prompts to open in new window

## Description
This PR adds a jump icon for onedrive links to the editor since onedrive links can never be embedded,

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-detail.js [hunk 1]
```diff
@@ -105,10 +107,21 @@ class ContentWebLinkDetail extends SkeletonMixin(ErrorHandlingMixin(LocalizeActi
...
 			let name = '';
+
 			if (webLinkEntity) {
 				name = webLinkEntity.type === CloudDriveTypes.OneDrive
 					? 'content.oneDriveLink'
 					: 'content.googleDriveLink';
+
+				if (webLinkEntity.type === CloudDriveTypes.OneDrive) {
+					defaultLinkPreview = html`
+						<d2l-activity-content-onedrive-link-jump-icon>
```
Comment (brissons) [inline components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-detail.js:118]: the designs show the jump icon further away from the banner. we might want to add some margins to space them out

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { css, html } from 'lit-element/lit-element.js';
+import { MobxLitElement } from '@adobe/lit-mobx';
+import { RtlMixin } from '@brightspace-ui/core/mixins/rtl-mixin.js';
```
Comment (brissons) [inline components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js:3]: not sure if this mixin will automatically flip the `svg` for rtl. we'll probably need to add some classes to makes sure this is the case.

Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js:3]: Interesting, didn't think we flipped images in RTL.

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+				}
+			`
+		];
+	}
+
+	render() {
+		return html`
+			<div class="d2l-jump-icon-outer-frame">
+				<div class="d2l-onedrive-jump-icon">
+					<svg xmlns="http://www.w3.org/2000/svg" width="252" height="200" viewBox="0 0 252 200">
```
[the remark this answers is not in the document]
Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js:26]: This is how LTI links should be displayed, so it's copied from that. In that case, maybe both need to be updated in another cleanup PR?

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+class ActivityContentOneDriveLinkJumpIcon extends RtlMixin(MobxLitElement) {
+	static get styles() {
+		return [
+			css`
+				.d2l-jump-icon-outer-frame {
+					align-items: center;
+					text-align: center;
+				}
+				.d2l-onedrive-jump-icon {
```
[the remark this answers is not in the document]
Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js:13]: This is how LTI links should be displayed, so it's copied from that. In that case, maybe both need to be updated in another cleanup PR?

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-detail.js [hunk 5]
```diff
@@ -105,10 +107,21 @@ class ContentWebLinkDetail extends SkeletonMixin(ErrorHandlingMixin(LocalizeActi
...
+							<p class="d2l-body-compact">
+								${this.localize('content.externalActivityOpened')}
```
[the remark this answers is not in the document]
Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-web-link-detail.js:120]: Grrr. Bad copy, nice catch.

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js (hunk 2 above)
Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js:3]: Oh, I didn't think it was pointing to anything in particular. Hmm, yeah that does make a lot of sense. There is another preview that also has a similar arrow now that I think of it.

Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js:3]: For sure, going to mark this as resolved for this PR, and make sure that the question is addressed and answered as part of the whole feature (not this PR - too many downstream PRs will need to get updated)

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (brissons) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js:26]: hmmm, it looks like the LTI icon is resizing properly (although it probably needs those min values). there must be something slightly different that the LTI icon is doing for resizing that this one isn't

[the remark this answers is not in the document]
Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js:26]: Interesting, can you post a quick screenshot and I will investigate

[the remark this answers is not in the document]
Comment (brissons) [reply components/d2l-activity-editor/d2l-activity-content-editor/web-link/d2l-activity-content-onedrive-link-jump-icon.js:26]: one-drive-resize
lti-resize

**Pull request merged**